ml-algorithms/src/main/java/org/opensearch/ml/engine/utils/ModelSerDeSer.java

```diff
@@ -85,13 +83,15 @@ public static byte[] serialize(Object model) {
     // This method has been tested in K-means, Linear Regression, Logistic regression, Anomaly Detection and Random Cut Forest summarization and passed.
     public static Object deserialize(byte[] modelBin) {
         try (ByteArrayInputStream inputStream = new ByteArrayInputStream(modelBin);
-             ValidatingObjectInputStream validatingObjectInputStream = new ValidatingObjectInputStream(inputStream)){
+             ValidatingObjectInputStream validatingObjectInputStream = new ValidatingObjectInputStream(inputStream);
+             ObjectInputStream objectInputStream = new ObjectInputStream(new ByteArrayInputStream(modelBin))){
```

Comment:
`ObjectInputStream objectInputStream = new ObjectInputStream(new ByteArrayInputStream(modelBin)))` =>

Can't we do this: `ObjectInputStream objectInputStream = new ObjectInputStream(inputStream)?